### ISSUE TITLE
Flutter renderer

### ISSUE THREAD

**Issue by ghost** [Jaguar-dart/nuts#2]
I see that the way you described Ribe a html offer tree feels very like flutter.

I was wondering is a html to flutter lib makes sense to build a flutter renderer.
There is one on GitHub and it pulls Wikipedia html, parses it and renders flutter objects.

Food for thought ?
It also is cool because you don't have to recompile

**Comment by tejainece** [comment]
I will take a look.

But we can do better. nuts is very abstracted from the widget library underneath. An implementation of Flutter renderer for ViewRenderer would suffice. Currently there is only implementation of HtmlRenderer. You can also do platform specific conditional rendering. For example, if platform is web, use this Component, elseif platform is Flutter, use this component, etc.

**Comment by ghost** [comment]
This is the one I was referring to.
https://pub.dartlang.org/packages/flutter_html_view

I have found its parser to be very fragile. But its more of a reference to see approaches i feel.

Wow there is conditional ? Like Vuejs has a formal DSL ? Very nice
approaches.

I presume your aiming for an agnostic layout so that the same layout works
on web and flutter ?

**Comment by tejainece** [comment]
> Wow there is conditional ? Like Vuejs has a formal DSL ?  

Yes. With some cool rx sugar added. There will XML template engine as well later. But I am really hoping this gets accepted: https://github.com/dart-lang/sdk/issues/34362

> I presume your aiming for an agnostic layout so that the same layout works on web and flutter ?  

Yes. Same UI tree targets different rendering backends. Sometimes though, one Component cannot directly target all platforms. In this case, you have to use platform specific Components based on conditional rendering.

**Comment by ghost** [comment]
Can you add setup to the readme ?
have not used webdev much for dart. Mostly only flutter.

EDIT: I just read your proposal. So for now this wont run without a custom build of dart ?

**Comment by tejainece** [comment]
> Can you add setup to the readme ?
> have not used webdev much for dart. Mostly only flutter.

Sure. Will do this by tomorrow.

> EDIT: I just read your proposal. So for now this wont run without a custom build of dart ?

What do you mean? This: https://github.com/dart-lang/sdk/issues/34362? No, That will make things cleaner to write. Has nothing directly to do with nuts.

**Comment by ghost** [comment]
Well the discussion in the main dart repo was talking about adding a new
operator to the dart language. Its the link you added above.

Thanks again. This is a interesting spin of an approach. Great potential. I
hope I can help.

**Comment by tejainece** [comment]
Here is an example boilerplate repo: https://github.com/nutsdart/boilerplate

It is still early stage. Suggestions are welcome. PRs will be awesome!

**Comment by ghost** [comment]
ok thanks. Will check it out.
